jetstream/statistics.py

```diff
@@ -231,54 +233,35 @@ def apply(
-            reference_branch = experiment.reference_branch
-            if reference_branch and reference_branch not in branch_list:
-                logger.warning(
-                    f"Branch {reference_branch} not in {branch_list} for {self.name()}.",
-                    extra={
-                        "experiment": experiment.normandy_slug,
-                        "metric": metric,
-                        "statistic": self.name(),
-                        "analysis_basis": analysis_basis.value,
-                        "segment": segment,
-                    },
-                )
-            else:
-                if reference_branch is None:
-                    ref_branch_list = branch_list
-                else:
-                    ref_branch_list = [reference_branch]
-
-                for ref_branch in ref_branch_list:
-                    try:
-                        statistic_result_collection.__root__.extend(
-                            self.transform(
-                                df,
-                                metric,
-                                ref_branch,
-                                experiment,
-                                analysis_basis,
-                                segment,
-                            ).__root__
-                        )
-                    except Exception as e:
-                        logger.exception(
+
+            for ref_branch in branch_list:
+                try:
+                    statistic_result_collection.__root__.extend(
+                        self.transform(
+                            df,
+                            metric,
+                            ref_branch,
+                            experiment,
+                            analysis_basis,
+                            segment,
+                        ).__root__
+                    )
+                except Exception as e:
+                    logger.exception(
+                        f"Error while computing statistic {self.name()} "
+                        + f"for metric {metric}: {e}",
+                        exc_info=StatisticComputationException(
                             f"Error while computing statistic {self.name()} "
-                            + f"for metric {metric}: {e}",
-                            exc_info=StatisticComputationException(
-                                f"Error while computing statistic {self.name()} "
-                                + f"for metric {metric}: {e}"
-                            ),
-                            extra={
-                                "experiment": experiment.normandy_slug,
-                                "metric": metric,
-                                "statistic": self.name(),
-                                "analysis_basis": analysis_basis.value,
-                                "segment": segment,
-                            },
-                        )
-
-                    df = df[df.branch != ref_branch]
```

Comment:
are we making sure somewhere downstream that we're not comparing a reference branch with itself?

Reply:
This only removes rows from the dataframe **after** a given reference branch statistic transform call, so removing it shouldn't affect any comparisons.

The only thing that it appears to do is prevent comparing branches in both directions (i.e., with each as reference branch for the other -- A vs B and B vs A), which we explicitly want to do with this PR.